android/app/src/main/cpp/audio_mixer.cpp

```
@@ -7,38 +7,73 @@
 
 void AudioMixer::updateDeviceAudio(int deviceId, const int16_t* audioData, int numFrames) {
-    std::lock_guard<std::mutex> lock(mixerMutex);
-    auto it = deviceBuffers.find(deviceId);
-    if (it != deviceBuffers.end()) {
-        it->second.assign(audioData, audioData + numFrames);
+    // Lock-free: find the device without locking the registry
+    // (assumes device pointers remain stable, which they do with std::unique_ptr in std::map)
+    DeviceAudioBuffer* device = nullptr;
+    {
+        std::lock_guard<std::mutex> lock(deviceRegistryMutex);
+        auto it = devices.find(deviceId);
```

Clang (14.0.6)

[warning] variable name 'it' is too short, expected at least 3 characters

(readability-identifier-length)